### DISCUSSION TITLE
Transform object for read after write operations

### DISCUSSION THREAD

**Discussion by onhate** [sensedeep/dynamodb-onetable#254] [General]
Hi, due to the limit on the record size I implemented a compressed field on my table , it should behave similar to crypt logic but compress on write and uncompress on reads, for that I used the table transform method with following logic:

```
transform(model: AnyModel, op: string, item: AnyEntity) {
  if (op === 'read') {
    if (item.content) {
      item.content = 'uncompressed content';
    }
  }
  if (op === 'write') {
    if (item.content) {
       item.content = 'compressed content';
    }
  }
  return item;
},
```

But I discovered that when I call `await model.create({  content: 'Very Large String' });` the returned value of the create method is the value parsed as `write` which means it returns the object with compressed string value, to circumvent that I need to perform a `get` after `create` so it returns the uncompressed value.

That makes me think if the returned value of `create`/`update` should also be parsed as read after executing the operation on database.

WDYT?

**Comment by mobsense** [comment]
This is a bug. I'll transfer into the issue list.

**Reply by onhate** [reply]
ok, good to know, thanks
